platforms/software/dependency-management/src/main/java/org/gradle/api/internal/catalog/LibrariesSourceGenerator.java

```diff
@@ -503,49 +503,48 @@ private void writeSingleVersionAccessor(String versionAlias, @Nullable String co
     }
 
     private void performValidation(List<String> libraries, List<String> bundles, List<String> versions, List<String> plugins) {
-        assertUnique(libraries, "library aliases", "");
-        assertUnique(bundles, "dependency bundles", "Bundle");
-        assertUnique(versions, "dependency versions", "Version");
-        assertUnique(plugins, "plugins", "Plugin");
+        assertUnique(problemService, libraries, "library aliases", "");
+        assertUnique(problemService, bundles, "dependency bundles", "Bundle");
+        assertUnique(problemService, versions, "dependency versions", "Version");
+        assertUnique(problemService, plugins, "plugins", "Plugin");
         int size = libraries.size() + bundles.size() + versions.size() + plugins.size();
         if (size > MAX_ENTRIES) {
-            throw throwVersionCatalogProblemException(problemService.create(builder ->
+            throw throwVersionCatalogProblemException(problemService.forCore().create(builder ->
                 configureVersionCatalogError(builder, getProblemPrefix() + "version catalog model contains too many entries (" + size + ").", TOO_MANY_ENTRIES)
                     .details("The maximum number of aliases in a catalog is " + MAX_ENTRIES)
                     .solution("Reduce the number of aliases defined in this catalog")
                     .solution("Split the catalog into multiple catalogs")));
         }
     }
 
-    private RuntimeException throwVersionCatalogProblemException(ReportableProblem problem) {
-        throw throwErrorWithNewProblemsApi(ERROR_HEADER, ImmutableList.of(problem));
+    private RuntimeException throwVersionCatalogProblemException(Problem problem) {
+        throw throwErrorWithNewProblemsApi(problemService, ERROR_HEADER, ImmutableList.of(problem));
```

Comment:
🤔 Seems like this `throwErrorWithNewProblemsApi` method should be part of the `InternalProblems` service, not a static method on a builder?